### PR TITLE
sched/mqueue: replace inline linklist to improve performance 

### PR DESCRIPTION
**Pull request by anchao**
## Summary

sched/mqueue: replace inline linklist to improve performance

## Impact

N/A

## Testing

mq_send test (cycle count)

```
434 (origin)
425 (fs/mqueue: skip nxmq_pollnotify() if no poll waiters)
415 (sched/mqueue: minor code tuning of message queue)
393 (sched/mqueue: replace inline linklist to improve performance)
361 (sched/mqueue: do sanity check if DEBUG_FEATURES is enabled)
```

### CODE REVIEW

File: sched/Kconfig [hunk 1]
```diff
@@ -1585,6 +1585,12 @@ config MQ_MAXMSGSIZE
 		Message structures are allocated with a fixed payload size given by this
 		setting (does not include other message structure overhead.
 
+config DISABLE_MQUEUE_NOTIFICATION
+	bool "Disable POSIX message queue notification"
+	default n
```
Comment (xiaoxiang781216) [inline sched/Kconfig:1590]: n->DEFAULT_SMALL

**Comment by xiaoxiang781216** [comment]
@anchao it's better to split the independent patch to different PR to boost review.

File: include/nuttx/mqueue.h [hunk 2]
```diff
@@ -29,6 +29,7 @@
 #include <nuttx/compiler.h>
 #include <nuttx/fs/fs.h>
 #include <nuttx/signal.h>
+#include <nuttx/list.h>
```
Comment (xiaoxiang781216) [inline include/nuttx/mqueue.h:32]: remove all #include <queue.h>

Comment (anchao) [reply include/nuttx/mqueue.h:32]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Please update PR description do match the actual changes

File: sched/mqueue/mq_initialize.c [hunk 3]
```diff
@@ -78,13 +61,13 @@ static struct mqueue_msg_s  *g_msgfreeirqalloc;
  *
  ****************************************************************************/
 
-static struct mqueue_msg_s *
-mq_msgblockalloc(FAR sq_queue_t *queue, uint16_t nmsgs,
+static void
+mq_msgblockalloc(FAR struct list_node *list, uint16_t nmsgs,
                  uint8_t alloc_type)
 {
   struct mqueue_msg_s *mqmsgblock;
```
Comment (pkarashchenko) [inline sched/mqueue/mq_initialize.c:68]:
```suggestion
   FAR struct mqueue_msg_s *mqmsgblock;
```

**Review by pkarashchenko — APPROVED** [review]

File: sched/mqueue/mq_initialize.c (hunk 3 above)
Comment (anchao) [reply sched/mqueue/mq_initialize.c]: Done!

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**